dfusion_rust_core/src/models/solution.rs

```diff
@@ -36,17 +37,24 @@ impl Serializable for Solution {
             .flat_map(|tup| once(tup.0).chain(once(tup.1)))
             .cloned()
             .collect();
-        [&self.prices, &alternating_buy_sell_amounts]
-            .iter()
-            .flat_map(|list| list.iter())
-            .flat_map(Serializable::bytes)
-            .collect()
+
+        [
+            &vec![self.prices.len() as u128],
+            &self.prices,
+            &alternating_buy_sell_amounts,
+        ]
+        .iter()
+        .flat_map(|list| list.iter())
+        .flat_map(Serializable::bytes)
+        .collect()
     }
 }
 
 impl Deserializable for Solution {
     fn from_bytes(mut bytes: Vec<u8>) -> Self {
-        let volumes = bytes.split_off(TOKENS as usize * 12);
+        // First 12 bytes encode the length of price vector (i.e. num_tokens)
+        let len_prices = U128::from_big_endian(bytes.drain(0..12).collect::<Vec<u8>>().as_slice());
```

Comment:
Can we use 2 bytes instead of 12?

Reply:
Yes sir! I had originally thought this would be a good idea, but didn't go with it immediately. Now that I have a second opinion! Will do.